pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/KitRequestShipping.java

```diff
@@ -1245,13 +1248,26 @@ public static String generateBspParticipantID(String collaboratorIdPrefix, Strin
         return null;
     }
 
-    public static String generateBspSampleID(@NonNull Connection conn, String collaboratorParticipantId, String type, int kitTypeId) {
+    public static String generateBspSampleID(@NonNull Connection conn, String collaboratorParticipantId, String type, int kitTypeId,
+                                             DDPInstance ddpInstance) {
         if (collaboratorParticipantId != null && collaboratorParticipantId.length() < COLLABORATOR_MAX_LENGTH) {
             String collaboratorSampleId = collaboratorParticipantId;
             if (StringUtils.isNotBlank(type)) {
                 collaboratorSampleId += "_" + type;
             }
+            String participantId = collaboratorParticipantId.split("_")[1];
             int counter = getKitCounter(conn, collaboratorSampleId, kitTypeId);
+
+            if  (ddpInstance.isMigratedDDP()) {
+                if (!ParticipantUtil.isHruid(participantId)) {
+                    //assuming participantId is a legacy short id
+                    counter += getPepperKitCounter(conn, collaboratorSampleId, participantId, kitTypeId,
+                            ddpInstance.getParticipantIndexES());
```

Comment:
For Migrated studies, if the legacy id is given, count the number of kits that were created with this participant's pepper short id and consider that number when creating the collaborator sample id for this kit. 

For example, if participant PABCDE has legacy short id 1, and has a saliva kit 0001_SALIVA_1 and PABCDE_SALIVA_2, the next kit should be 0001_SALIVA_3, and not 0001_SALIVA_2

Reply:
This comment is gold!  Please put it into source as a comment.